### PR TITLE
[Compute] BREAKING CHANGE: Migrate commands using Code Gen V2

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az disk create/update/grant-access`
`az image create/update`

**Description**

Migrate `disk` and `image` command group to Code Gen V2
Note: The subcommand groups have not yet been migrated

AAZ PR: https://github.com/Azure/aaz/pull/640

**Testing Guide**

**History Notes**

[Compute] BREAKING CHANGE: `az disk`: Refine output fields to align with backend service 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/vm/aaz/latest/disk/__cmd_group.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class __CMDGroup(AAZCommandGroup):
     """Manage Azure Managed Disks.
 
     Azure Virtual Machines use disks as a place to store an operating system, applications, and data. All Azure virtual machines have at least two disks: An operating system disk, and a temporary disk. The operating system disk is created from an image, and both the operating system disk and the image are actually virtual hard disks (VHDs) stored in an Azure storage account. Virtual machines also can have one or more data disks, that are also stored as VHDs.
-    Azure Unmanaged Data Disks have a maximum size of 4095 GB. To use disks larger than 4095 GB use [Azure Managed Disks](https://learn.microsoft.com/azure/virtual-machines/managed-disks-overview)
+    Azure Unmanaged Data Disks have a maximum size of 4095 GB. To use disks larger than 4095 GB use [Azure Managed Disks](https://docs.microsoft.com/azure/virtual-machines/managed-disks-overview)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/latest/disk/__cmd_group.py:21]: Why do we need to modify the URL here?

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/latest/disk/__cmd_group.py:21]: You are right. We don't need to change the URL.
I have no idea how it was changed. I'll change it back.

File: src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2018_03_01_hybrid/disk/__cmd_group.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ class __CMDGroup(AAZCommandGroup):
     """Manage Azure Managed Disks.
 
     Azure Virtual Machines use disks as a place to store an operating system, applications, and data. All Azure virtual machines have at least two disks: An operating system disk, and a temporary disk. The operating system disk is created from an image, and both the operating system disk and the image are actually virtual hard disks (VHDs) stored in an Azure storage account. Virtual machines also can have one or more data disks, that are also stored as VHDs.
-    Azure Unmanaged Data Disks have a maximum size of 4095 GB. To use disks larger than 4095 GB use [Azure Managed Disks](https://learn.microsoft.com/azure/virtual-machines/managed-disks-overview)
+    Azure Unmanaged Data Disks have a maximum size of 4095 GB. To use disks larger than 4095 GB use [Azure Managed Disks](https://docs.microsoft.com/azure/virtual-machines/managed-disks-overview)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2018_03_01_hybrid/disk/__cmd_group.py]: same as above

File: src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2019_03_01_hybrid/disk/__cmd_group.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ class __CMDGroup(AAZCommandGroup):
     """Manage Azure Managed Disks.
 
     Azure Virtual Machines use disks as a place to store an operating system, applications, and data. All Azure virtual machines have at least two disks: An operating system disk, and a temporary disk. The operating system disk is created from an image, and both the operating system disk and the image are actually virtual hard disks (VHDs) stored in an Azure storage account. Virtual machines also can have one or more data disks, that are also stored as VHDs.
-    Azure Unmanaged Data Disks have a maximum size of 4095 GB. To use disks larger than 4095 GB use [Azure Managed Disks](https://learn.microsoft.com/azure/virtual-machines/managed-disks-overview)
+    Azure Unmanaged Data Disks have a maximum size of 4095 GB. To use disks larger than 4095 GB use [Azure Managed Disks](https://docs.microsoft.com/azure/virtual-machines/managed-disks-overview)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2019_03_01_hybrid/disk/__cmd_group.py]: same as above

File: src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2018_03_01_hybrid/image/_show.py [hunk 4]
```diff
@@ -193,7 +193,6 @@ def _build_schema_on_200(cls):
             )
             storage_profile.os_disk = AAZObjectType(
                 serialized_name="osDisk",
-                flags={"required": True},
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2018_03_01_hybrid/image/_show.py:196]: Why do we need to remove it?

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 5]
```diff
@@ -10575,24 +10574,24 @@ def test_vm_trusted_launch_os_disk_secure_upload(self):
...
 
         self.cmd('disk create -n {disk1} -g {rg} --os-type Windows --hyper-v-generation v2 --security-type TrustedLaunch --upload-type UploadWithSecurityData --upload-size-bytes 34359738880 --sku standard_lrs', checks=[
             self.check('securityProfile.securityType', 'TrustedLaunch'),
             self.check('creationData.createOption', 'UploadPreparedSecure')
         ])
         self.cmd('disk grant-access -n {disk1} -g {rg} --access-level Write --duration-in-seconds 86400 --secure-vm-guest-state-sas', checks=[
-            self.exists('accessSas'),
-            self.exists('securityDataAccessSas')
+            self.exists('accessSAS'),
+            self.exists('securityDataAccessSAS')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:10587]: Why do many attribute names change? Are they breaking changes for customers?

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:10587]: It is caused by deserialization of code Gen V2.
@kairu-ms could you please comment on this?

File: src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2018_03_01_hybrid/image/_show.py (hunk 4 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/profile_2018_03_01_hybrid/image/_show.py:196]: It wasn't supposed to exist. Now just fix it.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 5 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:10587]: Got it, thank you~ So they are breaking changes for customers, right? If so, do we need to add a [BREAKING CHANGE] tag to make customers aware of these breaking changes?

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:10587]: Added a history note with "BREAKING CHANGE" marker:

image

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:10587]: image
Please refine PR title as well

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**